bucket_map/src/index_entry.rs

```diff
@@ -19,30 +19,39 @@ use {
 pub struct IndexEntry {
     pub key: Pubkey, // can this be smaller if we have reduced the keys into buckets already?
     pub ref_count: RefCount, // can this be smaller? Do we ever need more than 4B refcounts?
-    storage_offset: u64, // smaller? since these are variably sized, this could get tricky. well, actually accountinfo is not variable sized...
+    // storage_offset_mask contains both storage_offset and storage_capacity_when_created_pow2
+    // see _MASK_ constants below
+    storage_offset_mask: u64, // smaller? since these are variably sized, this could get tricky. well, actually accountinfo is not variable sized...
     // if the bucket doubled, the index can be recomputed using create_bucket_capacity_pow2
-    storage_capacity_when_created_pow2: u8, // see data_location
     pub num_slots: Slot, // can this be smaller? epoch size should ~ be the max len. this is the num elements in the slot list
 }
 
+/// how many bits to shift the capacity value in the mask
+const STORAGE_OFFSET_MASK_CAPACITY_SHIFT: u64 =
+    ((std::mem::size_of::<u64>() - std::mem::size_of::<u8>()) * u8::BITS as usize) as u64;
```

Comment:
How about this instead?

```suggestion
const STORAGE_OFFSET_MASK_CAPACITY_SHIFT: u64 = (u64::BITS - u8::BITS) as u64;
```